docs/src/main/asciidoc/cyclonedx.adoc

```diff
@@ -0,0 +1,183 @@
+
+=== Maven Dependency SBOMs
+
+For Quarkus Maven projects dependency SBOMs can be generated with `quarkus:dependency-sbom` goal. The outcome of the goal will be saved in a `target/<artifactId>-<version>-dependency-cyclonedx.json` file (which can be changed by setting `outputFile` goal parameter or `quarkus.dependency.sbom.output-file` property). The complete Quarkus build and runtime dependency graphs will be recoded in the https://cyclonedx.org/[CycloneDX] `JSON` format.
+
+`XML` format can be requested by setting `format` goal parameter (or `quarkus.dependency.sbom.format` property) to `xml`.
+
+Each component in the generated SBOM will include `quarkus:component:scope` property that will indicate whether this component is used at runtime or only development/build time.
+[source,json]
+----
+        {
+          "name" : "quarkus:component:scope",
+          "value" : "runtime"
+        }
+----
+
+By default, `quarkus:dependency-sbom` captures the dependencies of a production build. Quarkus supports three application bootstrap modes: normal (production), test and dev. In all three modes an application may have different dependency graphs. `mode` parameter can be used to indicate which dependency graph should be recorded. If `mode` is set to `test` or `dev`, the target file name will become `target/<artifactId>-<version>-<mode>-dependency-cyclonedx.json`.
+
+The complete set of parameters and their description can be obtained by executing `mvn help:describe -Dcmd=quarkus:dependency-sbom -Ddetail`.
+
+=== Gradle Dependency SBOMs
+
+Unlike Maven, the https://github.com/CycloneDX/cyclonedx-gradle-plugin[Gradle CycloneDX plugin implementation] can be used in Quarkus projects to generate dependency SBOMs, since the implementation manifests dependency configurations registered by configured plugins.
+
+Please, refer to the https://github.com/CycloneDX/cyclonedx-gradle-plugin[Gradle CycloneDX plugin] documentation for its configuration options. Here is a list of Quarkus dependency configurations that would be relevant for manifesting:
+
+* `quarkusProdRuntimeClasspathConfiguration` - Quarkus application production runtime dependencies;
+* `quarkusProdRuntimeClasspathConfigurationDeployment` - Quarkus application production runtime and build time dependencies;
+* `quarkusTestRuntimeClasspathConfiguration` - Quarkus application test runtime dependencies;
+* `quarkusTestRuntimeClasspathConfigurationDeployment` - Quarkus application test runtime and build time dependencies;
+* `quarkusDevRuntimeClasspathConfiguration` - Quarkus application dev mode runtime dependencies;
+* `quarkusDevRuntimeClasspathConfigurationDeployment` - Quarkus application dev mode runtime and build time dependencies.
+
+Given that the plugin is not aware of how Quarkus uses these dependencies, it will not be able to set `quarkus:component:scope` property for components. On the other hand, the requested configuration name can be used indicate which scope to target.
```

Comment:
```suggestion
Given that the plugin is not aware of how Quarkus uses these dependencies, it will not be able to set the `quarkus:component:scope` property for components. On the other hand, the requested configuration name can be used indicate which scope to target.
```

Reply:
Yes because here it's definite, it's this particular property you want to set.